### PR TITLE
29 Encode picture numbers correctly

### LINKED ISSUE #27: Mixed line endings
Line endings are a mix of LF and CRLF throughout the repository - it would be nice if these were tidied up.

### LINKED ISSUE #29: `Encode*` does not set picture numbers correctly
Picture numbers must have consecutive, ascending integer values, wrapping at (2**32)-1 back to 0.

Using EncodeHQ-ConstQ to encode the _real_pictures_ file with

    EncodeHQ-ConstQ \
        -x 1280 -y 720 -f 4:2:2 -z 10 -r 6 \
        -k LeGall -d 3 -a 2 -u 1 -v -q 0 -S 2 \
       ./real_pictures.16p2 ./encoded_real_pictures.vc2

gives a non-conformat bitstream with sequential pictures having a picture number of zero. It also gives a nonconformant stream when the interlaced flag (`-i`) is set

Encode_CBR produces nonconformant bitstream with the same error if the interlaced flag is set (but not otherwise).

### PR DESCRIPTION
**Pull request by GalenReich**
Closes #29 and produces conformant bitstreams from the sample code.

Writes pictures (frames or fields if interlaced) with consecutive, ascending integer values, wrapping at (2**32)-1 back to 0.
Changes types of some parameters to work when frame numbers exceed 2^32-1.
Adds test for picture number utility function.

Whitespace changes to line endings close #27 (everything to LF from mix of CRLF and LF)

### CODE REVIEW

File: src/EncodeHQ-CBR/EncodeHQ-CBR.cpp [hunk 1]
```diff
@@ -285,7 +285,7 @@ try { //Giant try block around all code to get error messages
   Array2D uDiff(format.chromaShape()); // used for PSNR calculation only
   Array2D vDiff(format.chromaShape()); // used for PSNR calculation only
 
-  int frame = 0;
+  unsigned long long frame = 0;
```
Comment (jamesba) [inline src/EncodeHQ-CBR/EncodeHQ-CBR.cpp:288]: At some point if you intend to maintain this codebase I would suggest including `stdint` and using sized ints in place of these. Technically the size of types like `unsigned long long int` are archetecture dependent.

File: src/Library/src/Utils.cpp [hunk 2]
```diff
@@ -47,6 +47,15 @@ const int utils::intlog2(int value) {
   return log;
 }
 
+// Calculate the picture number given a frame number, picture number, and number 
+// of pictures per frame - 1 or 2 (interlaced)
+unsigned long utils::getPictureNumber(int fieldNumber, unsigned long long frameNumber, const int fieldsPerFrame){
+    unsigned long long bigPictureNumber = fieldNumber + frameNumber*fieldsPerFrame;
```
Comment (jamesba) [inline src/Library/src/Utils.cpp]:
```suggestion
    const unsigned long long bigPictureNumber = fieldNumber + frameNumber*fieldsPerFrame;
```

File: src/Library/src/Utils.cpp [hunk 3]
```diff
@@ -47,6 +47,15 @@ const int utils::intlog2(int value) {
...
+    const unsigned long long bitLimit32 = 4294967296; // 2^32
```
Comment (jamesba) [inline src/Library/src/Utils.cpp]:
```suggestion
    const unsigned long long bitLimit32 = (1ULL) << 32;
```

File: src/EncodeHQ-CBR/EncodeHQ-CBR.cpp (hunk 1 above)
Comment (jamesba) [reply src/EncodeHQ-CBR/EncodeHQ-CBR.cpp:288]: Doesn't need to be now, though

Comment (GalenReich) [reply src/EncodeHQ-CBR/EncodeHQ-CBR.cpp:288]: Okay, that's good to know, I'll open an issue so this doesn't get forgotten :)

**Pull request merged**